growthbook.py

```diff
@@ -1023,65 +1278,93 @@ def _run(self, experiment: Experiment, featureId: Optional[str] = None) -> Resul
         if experiment.status == "draft" or not experiment.active:
             logger.debug("Experiment %s is not active, skip", experiment.key)
             return self._getExperimentResult(experiment, featureId=featureId)
+        
         # 6. Get the user hash attribute and value
-        hashAttribute = experiment.hashAttribute or "id"
-        hashValue = self._getHashValue(hashAttribute)
+        (hashAttribute, hashValue) = self._getHashValue(experiment.hashAttribute, experiment.fallbackAttribute)
         if not hashValue:
             logger.debug(
                 "Skip experiment %s because user's hashAttribute value is empty",
                 experiment.key,
             )
             return self._getExperimentResult(experiment, featureId=featureId)
 
-        # 7. Filtered out / not in namespace
-        if experiment.filters:
-            if self._isFilteredOut(experiment.filters):
-                logger.debug(
-                    "Skip experiment %s because of filters/namespaces", experiment.key
-                )
-                return self._getExperimentResult(experiment, featureId=featureId)
-        elif experiment.namespace and not inNamespace(hashValue, experiment.namespace):
-            logger.debug("Skip experiment %s because of namespace", experiment.key)
-            return self._getExperimentResult(experiment, featureId=featureId)
-
-        # 7.5. If experiment has an include property
-        if experiment.include:
-            try:
-                if not experiment.include():
+        assigned = -1
+
+        found_sticky_bucket = False
+        sticky_bucket_version_is_blocked = False
+        if self.sticky_bucket_service and not experiment.disableStickyBucketing:
+            sticky_bucket = self._get_sticky_bucket_variation(experiment.key, experiment.bucketVersion, experiment.minBucketVersion, experiment.meta)
+            found_sticky_bucket = sticky_bucket.get('variation', 0) >= 0
+            assigned = sticky_bucket.get('variation', 0)
+            sticky_bucket_version_is_blocked = sticky_bucket.get('versionIsBlocked', False)
+        
+        if found_sticky_bucket:
+            logger.debug("Found sticky bucket for experiment %s, assigning sticky variation %s", experiment.key, assigned)
```

Comment:
remove?

Reply:
I think this is fine to leave as a debug log.  It is helpful if trying to figure out why a variation was assigned.